hoot-services/src/main/java/hoot/services/controllers/job/DeriveChangesetResource.java

```diff
@@ -0,0 +1,164 @@
+        
+        String jobId = UUID.randomUUID().toString();
+        jobId = "ex_" + jobId.replace("-", "");
+
+        try {
+            JSONArray commandArgs = new JSONArray();
+            JSONObject arg = new JSONObject();
+            
+            //assuming that this would only be used with data layers already being loaded into the
+            //database
+            arg.put("input1", DB_URL + "/" + input1);
+            arg.put("input2", DB_URL + "/" + input2);
+            arg.put("aoi", aoi);
+            //write the changeset to a temp file which can be retrieved through this class
+            File tempOutputDir = new File(TEMP_OUTPUT_PATH);
+            if (!tempOutputDir.exists())
+            {
+              tempOutputDir.mkdir();   
+            }
+            //don't auto-delete this temp file, since we want it to be available for future access;
+            //users will be responsible for manually clearing this out of the hoot temp folder
+            //services always write changeset with xml; no use case for the sql right now
+            arg.put("output", File.createTempFile("changeset-", ".osc", tempOutputDir).getAbsolutePath());
+            arg.put("jobId", jobId);
+            commandArgs.add(arg);
+            
+            postJobRequest(jobId, createPostBody(commandArgs));
+        }
+        catch (Exception e) {
+            throw new WebApplicationException(e, Response.serverError().entity("Error deriving changeset for inputs: " + input1 + ", " + input2).build());
+        }
+
+        return new JobId(jobId);
+    }
+    
+    public String getJobStatusDetail(String jobId) throws Exception
+    {
+        JobStatus jobStatus = (new JobStatusManager()).getJobStatusObj(jobId);
+        if (jobStatus == null)
+        {
+            throw new IOException("Job with ID: " + jobId + " does not exist.");
+        }
+        return jobStatus.getStatusDetail(); 
+    }
+
+    /**
+     * Returns the contents of an XML changeset file (.osc)
+     * 
+     * @param jobId job ID the changeset file is associated with
+     * @return changeset XML contents
+     * @throws WebApplicationException if the job with ID = jobID does not exist, the referenced
+     * temp changeset file no longer exists, or the changeset is made up of multiple changeset 
+     * files.
+     */
+    @GET
+    @Path("/getderived")
+    @Produces(MediaType.TEXT_XML)
+    public Response get(@PathParam("jobId") String jobId) {
+        
+        Document responseDoc = null;
+        String changesetFileName = null;
+        try {
+            changesetFileName = getJobStatusDetail(jobId);
+            if (changesetFileName.equals("<multiple files>"))
+            {
+                throw new Exception("Changeset requested is made up of multiple changesets in multiple files.  /getderived does not currently support this.");
+            }
```

Comment:
I see we just throw an exception when multiple changeset files are necessary.  What's the threshold on changeset size?